.github/workflows/intermittent-test-check.yml

```diff
@@ -147,6 +147,8 @@ jobs:
     steps:
       - name: Checkout project
         uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.inputs.ref }}
```

Comment:
Since the run-test job uses the right ref, the issue occurs only when build and test versions are far apart I guess

Reply:
Yes, that's how we found it, since recently version was bumped to `2.1.0-SNAPSHOT`.